packages/xsnap/src/xsnap.c

```diff
@@ -204,7 +218,7 @@ int main(int argc, char* argv[])
 	xsCreation* creation = &_creation;
 
 	txSnapshot snapshot = {
-		"xsnap 0.1.0",
+		"xsnap-" XSNAP_VERSION,
```

Comment:
That's a significant compatibility policy we're baking into the snapshot format right there. It means we're claiming 1: we won't make any snapshot-compatibility-breaking changes without also changing the package.json version, and 2: every package.json version change means a compatibility-breaking change. This sort of binding between "marketing" version and compatibility version means we cannot commit a change to the XS code without simultaneously making a release (otherwise we'd be claiming compatibility for the tree before and after the XS commit).

If we're going to move away from a fixed `0.1.0` now, I'd recommend introducing a separate "snapshot compatibility version" (a sequential integer, not a semver string), instead of tying it to the package.json version. (Storing that snapshot compatibility version under a separate key in `package.json` is a fine idea, assuming it doesn't conflict with anything else that parses the file.. I can't think of a better place to put it).

Reply:
I've introduced a `SNAPSHOT_SIGNATURE` manifest constant.